core/domain/exp_domain.py

```diff
@@ -1776,6 +1776,67 @@ def create_default_state(
                 default_dest_state_name),
             feconf.DEFAULT_CONTENT_IDS_TO_AUDIO_TRANSLATIONS)
 
+    @classmethod
+    def convert_state(cls, state_dict, conversion_fn):
+        """Applies a conversion function on all the html strings in a state
+        to migrate them to a desired state.
+        Args:
+            state_dict: dict. The dict representation of State object.
+            conversion_fn: function. The conversion function to be applied on
+                the states_dict.
+
+        Returns:
+            dict. The converted state_dict.
+        """
+        state_dict['content']['html'] = (
+            conversion_fn(
+                state_dict['content']['html']))
+        if state_dict['interaction']['default_outcome']:
+            interaction_feedback_html = state_dict[
+                'interaction']['default_outcome']['feedback']['html']
+            state_dict['interaction']['default_outcome']['feedback'][
+                'html'] = conversion_fn(
+                    interaction_feedback_html)
+
+        for answer_group_index, answer_group in enumerate(
+                state_dict['interaction']['answer_groups']):
+            answer_group_html = answer_group['outcome']['feedback']['html']
+            state_dict['interaction']['answer_groups'][
+                answer_group_index]['outcome']['feedback']['html'] = (
+                    conversion_fn(
+                        answer_group_html))
+            if state_dict['interaction']['id'] == 'ItemSelectionInput':
+                for rule_spec_index, rule_spec in enumerate(
+                        answer_group['rule_specs']):
+                    for x_index, x in enumerate(rule_spec['inputs']['x']):
+                        state_dict['interaction']['answer_groups'][
+                            answer_group_index]['rule_specs'][
+                                rule_spec_index]['inputs']['x'][x_index] = (
+                                    conversion_fn(x))
+        for hint_index, hint in enumerate(
+                state_dict['interaction']['hints']):
+            hint_html = hint['hint_content']['html']
+            state_dict['interaction']['hints'][hint_index][
+                'hint_content']['html'] = (
+                    conversion_fn(hint_html))
```

Comment:
Move to previous line and then remove the outer parens.

Reply:
Done!